### ISSUE TITLE
Role deduplication and dependency variable

### ISSUE THREAD

**Issue by jbguerraz** [ansible/ansible#68465]
##### SUMMARY

Actually, the same than https://github.com/ansible/ansible/issues/12634 raised by https://github.com/ansible/ansible/issues/68463#issuecomment-603962555

Pasting summary from #12634 for confort:

Ansible executes roles with dependencies in an unexpected way. Given the following:
A role called common which accepts a parameter.
A role called role1 which depends on common (the dependency is declared through the role's
meta/main.yml file). It also takes a parameter which it passes down to the common role via
variable expansion.
A role called role2 which does the same as above.
The site.yml file which "calls" both role1 and role2 passing a parameter with two
different values.
It seems that Ansible optimizes execution of dependent roles when they are passed the same set of
parameters by running the dependency only once. It seems to do so before variable expansion takes
place, which means that with the following files, Ansible would execute the common role only once even though, at runtime, we expect it to be executed twice, since the parameter is expanded to two different values.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

roles

##### ANSIBLE VERSION

```paste below
ansible 2.8.5
  config file = None
  configured module search path = ['~/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib64/python3.6/site-packages/ansible
  executable location = /usr/lib/python-exec/python3.6/ansible
  python version = 3.6.9 (default, Nov  4 2019, 05:05:26) [GCC 8.2.0]
```

##### CONFIGURATION

```paste below
None
```

##### OS / ENVIRONMENT

Linux (Gentoo)

##### STEPS TO REPRODUCE

run https://github.com/jbguerraz/ansibledependencies

```yaml
git clone https://github.com/jbguerraz/ansibledependencies
cd ansibledependencies
sed -i 's/allow_duplicates: true/allow_duplicates: false/' roles/library-builder/meta/main.yaml
sed -i 's/allow_duplicates: true/allow_duplicates: false/' roles/image-builder/meta/main.yaml
ansible-playbook playbook.yaml
```

##### EXPECTED RESULTS

Variable is substituted before role deduplication logic

##### ACTUAL RESULTS

Comparison is done with variable not substituted, role is then wrongly deduplicated.

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/playbook/role`](https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/role)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sivel** [comment]
Duplicate of:

* https://github.com/ansible/ansible/issues/72650
* https://github.com/ansible/ansible/issues/67718
